### ISSUE TITLE
`#` parsing inhibits use of some commands

### ISSUE THREAD

**Issue by Lord-Valen** [nushell/nushell#6335]
### Describe the bug

This is a blocker for nix users. `#` is used to indicate comments in nushell but some commands use it.
For example with nix-commands you might want to:
```shell
nix run nixpkgs#cowsay "I'm a cow!"
```
Or:
```shell
nix profile install nixpkgs#cowsay
```
But nushell renders this useless as everything after `#` is a comment.

### How to reproduce

1. Be using nushell
2. Have nix installed
3. Have nix-commands enabled
4. `nix run nixpkgs#<somePackage>`

### Expected behavior

I expected nu to interpret `#` as part of the command if there is not whitespace in front.

### Screenshots

![image](https://user-images.githubusercontent.com/46138807/184688160-3f4981da-9fbb-4b10-adaa-f7e1f81f0175.png)

### Configuration

| key                | value                                           |
| ------------------ | ----------------------------------------------- |
| version            | 0.65.0                                          |
| branch             |                                               |
| commit_hash        |                                                 |
| build_os           | linux-x86_64                                    |
| build_target       | x86_64-unknown-linux-gnu                        |
| rust_version       | rustc 1.62.1                                    |
| cargo_version      | cargo 1.62.0                                    |
| pkg_version        | 0.65.0                                          |
| build_time         | 1980-01-01 00:00:00 +00:00                      |
| build_rust_channel | release                                         |
| features           | database, dataframe, default, trash, which, zip |
| installed_plugins  |                                                 |

**Comment by jmoore34** [comment]
Does `nix run 'nixpkgs#cowsay' "I'm a cow"` or `nix profile install 'nixpkgs#cowsay'` work?

**Comment by kubouch** [comment]
You should be able to call external commands with `#` like this:
```
^`foo#bar`
```
but this shouldn't be necessary. What @jmoore34 suggested should work.

As a side note, you can create a custom command with `#`
```
def `foo#bar` [] { }
```
but this shouldn't be possible, we need to fix the parser to not allow it when checking the command name.

**Comment by Lord-Valen** [comment]
> Does `nix run 'nixpkgs#cowsay' "I'm a cow"` or `nix profile install 'nixpkgs#cowsay'` work?

Ah yes, it does. Thanks! But there's still a question of whether the default behaviour should change anyway.

**Comment by sholderbach** [comment]
Resolved by #8151
